client/v3/client.go

```diff
@@ -475,6 +476,14 @@ func (c *Client) roundRobinQuorumBackoff(waitBetween time.Duration, jitterFracti
 	}
 }
 
+func minSupportedVersion() *semver.Version {
+	ver := semver.Must(semver.NewVersion(version.Version))
+	if ver.Minor == 0 {
```

Comment:
This is a breaking change. Applications based on `clientv3 SDK 4.0` can't talk to etcdserver of previous version (e.g. 3.7, assuming there is no 3.8) if `RejectOldCluster` is true. Obviously the versions (4.0 and 3.7) should be compatible.

Two proposals:
- Clearly document the behaviour change (at least in Changelog). Since `RejectOldCluster` is false by default, it might not be a big problem.
- Always figure out the correct previous minor release no matter current version is the first minor version (e.g. 4.0) or not, and allow client with current version to connect to it. Since we already define a list a version in [version.go#L38-L46](https://github.com/etcd-io/etcd/blob/af07f18dce96d4418bc1e89710675a44e3e5028b/api/version/version.go#L38-L46). We can add them into a slice, then that should be easy enough to do the trick. [**preferred**]

Reply:
Not sure how you define a breaking change, but we didn't release v4.0 so don't think we are breaking anyone. When we release v4.0, we can decide whether and how many v3.X minor etcd version we want to support. Please remember that when we will cut v4.0, it will be using different code `client/v4` and not `client/v3` :P.

Reply:
>  but we didn't release v4.0 so don't think we are breaking anyone.

- Usually we don't add breaking change in a patch release, so no breaking change are breaking anyone at current stage.
- It's a foreseeable behaviour change. Why don't we handle it now with just minor effort? We already handle similar case in [test.sh#L571-L576](https://github.com/etcd-io/etcd/blob/6981e4820fb9ab286d1cb3effc141987d88f8de9/scripts/test.sh#L571-L576) 
- Another approach is to just add a TODO, similar to [downgrade.go#L59](https://github.com/etcd-io/etcd/blob/6981e4820fb9ab286d1cb3effc141987d88f8de9/server/etcdserver/version/downgrade.go#L59). But I think we should update it in a similar way as I mentioned in [above](https://github.com/etcd-io/etcd/pull/16313#discussion_r1275941354).

Overall it isn't a big problem. Either follow my [comment above](https://github.com/etcd-io/etcd/pull/16313#discussion_r1275941354) to make the change, or raise a ticket to track all such changes in 4.0.

> we can decide whether and how many v3.X minor etcd version we want to support

We have already clearly called out that we only support two releases in https://etcd.io/docs/v3.6/op-guide/versioning/

> it will be using different code `client/v4` and not `client/v3`

It's true. But most likely we will just copy the files under client/v3 to client/v4 firstly, and will carry over all legacy potential issues.

Reply:
@CaojiamingAlan Please raise a followup ticket as I mentioned [above](https://github.com/etcd-io/etcd/pull/16313#discussion_r1276190616) . Afterwards, I will approve & merge this PR.